src/main/java/org/folio/auth/authtokenmodule/tokens/legacy/EnhancedSecurityTenants.java

```diff
@@ -0,0 +1,35 @@
+package org.folio.auth.authtokenmodule.tokens.legacy;
+
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+
+public class EnhancedSecurityTenants {
+  public static final String ENHANCED_SECURITY_TENANTS = "enhanced.security.tenants";
+
+  public static final String ENHANCED_SECURITY_TENANTS_ENV = "ENHANCED_SECURITY_TENANTS";
+
+  public boolean isEnhancedSecurityTenant(String tenantId) {
+    return tenants.contains(tenantId);
+  }
+  private final List<String> tenants;
+  public EnhancedSecurityTenants() {
+    this.tenants = parseTenants();
+  }
+
+  private List<String> parseTenants() {
+    var prop = getTenantsFromEnvOrSystemProperty();
+    if (prop == null) {
+      return Collections.emptyList();
+    }
+    return Arrays.asList(prop.replace(" ", "").split(","));
+  }
+
+  private String getTenantsFromEnvOrSystemProperty() {
+    var prop = System.getProperty(ENHANCED_SECURITY_TENANTS);
+    if (prop != null) {
```

Comment:
better to use StringUtils.isNotBlank(prop) here, and the same in previous method, because someone can path empty string that is not valid case for us

Reply:
This is a good catch Serhii. I decided to check for the empty string and then throw an exception just in case people think that having the environment variable key is enough to enable enhanced security mode for all tenants. Although IDK maybe that should be a feature of this, to have a mechanism to enable it for all tenants...